example/basic/basic_hcube.cxx

```diff
@@ -26,81 +26,137 @@
 #include <t8_forest.h>
 #include <t8_cmesh_vtk.h>
 
+/** Construct the cmesh given the dimension of the examples
+ * \param[in] dim           The dimension of the example
+ * \param[in] do_partition  Option to partition the cmesh  */
+t8_cmesh_t
```

Comment:
```suggestion
static t8_cmesh_t
```
Functions that are not declared in a header/not used in another file should be static